atc/builds/tracker.go

```diff
@@ -38,9 +43,12 @@ func (bt *Tracker) Run(ctx context.Context) error {
 	builds, err := bt.buildFactory.GetAllStartedBuilds()
 	if err != nil {
 		logger.Error("failed-to-lookup-started-builds", err)
-		return err
+		builds = []db.Build{}
 	}
 
+	// TODO: maybe use a go routine to receive builds from ch.
+	builds = append(builds, bt.fetchInMemoryBuild(logger)...)
```

Comment:
@clarafu brought up a good point here - if there are multiple ATCs, we aren't guaranteed that lidar and the tracker will run on the same ATC. So we could end up queueing a bunch of in-memory checks on ATC1 (since ATC1 keeps acquiring the lock for lidar), but ATC2 keeps acquiring the lock for the tracker so ATC1 can't empty its queue of checks. This could result in checks taking a really long time to run.

What if we had a separate "check tracker" thing that runs on each ATC without a DB lock. It could just be a goroutine that runs either periodically (or can be signalled by lidar for immediate checking) that does effectively the same thing as `builds/tracker.go`, but uses the in memory builds rather than `BuildFactory.GetAllStartedBuilds`

Reply:
Yeah, what I'm going to do in next step.